Update CONTRIBUTING.md with coding standards and best practices (#49)
dermatz/ddev-woodoo-buildtools-magento: CONTRIBUTING.md

## General

> -   Make sure to have a working local development environment of [ddev](https://github.com/ddev/ddev).
> -   Make sure to have a working local development environment, e.g. ddev

Reviewer: ```suggestion
-   Make sure to have a working local development environment of [DDEV](https://github.com/ddev/ddev).
```

---

> ## VSCODE USERS
> 
> VSCode User automaticly use our VSCode Workspace Settings, located in `.vscode/settings.json`.
> We support [Git commit message helper](https://marketplace.visualstudio.com/items?itemName=D3skdev.git-commit-message-helper) to add
> correct commit message with prefixes based on your branch name given from github with issue-id.
> e.G. `#123 - Commit Message ...`
> 
> ![Demo Git commit message helper](https://github.com/d3skdev/git-prefix/raw/master/images/demo.gif)

Reviewer: ```suggestion
## VSCode Users

VSCode users automatically apply the workspace settings defined in `.vscode/settings.json`. 

To streamline your workflow, we recommend using the [Git Commit Message Helper](https://marketplace.visualstudio.com/items?itemName=D3skdev.git-commit-message-helper) extension. This tool ensures your commit messages are formatted correctly with the appropriate prefixes, based on the branch name and issue ID from GitHub.

For example: `#123 - Commit Message ...`

![Git Commit Message Helper Demo](https://github.com/d3skdev/git-prefix/raw/master/images/demo.gif)
```

---

## Best Practices

-   Write clear and concise comments

> -   Avoid global variables, use `local VARIABLE` if possible

Reviewer: ```suggestion
-   Avoid global variables, use `local VARIABLE` whenever possible
```
